lib/child_process.js

```diff
@@ -59,24 +59,42 @@ function stdioStringToArray(option) {
   }
 }
 
-exports.fork = function fork(modulePath /* , args, options */) {
+function normalizeForkArguments(args, options) {
+  if (Array.isArray(args)) {
+    args = args.slice(0);
+  } else if (options !== undefined &&
+             (args === undefined ||
+              args === null ||
+              typeof args !== 'object')
+  ) {
+    // throw new ERR_INVALID_ARG_TYPE('args', 'object', args);
+    args = [];
+  } else {
+    options = args;
+    args = [];
+  }
 
-  // Get options and args arguments.
-  var execArgv;
-  var options = {};
-  var args = [];
-  var pos = 1;
-  if (pos < arguments.length && Array.isArray(arguments[pos])) {
-    args = arguments[pos++];
+  if (options === undefined) {
+    options = {};
+  } else if (typeof options !== 'object') {
+    throw new ERR_INVALID_ARG_TYPE('options', 'object', options);
+  } else {
+    options = {};
```

Comment:
This is overwriting `options` if it is already an object or is `null`. My guess is this block is only meant for the latter case?

If so, perhaps this block should be removed completely and `|| options === null` added to the first conditional that sets `options = {}`?